packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -737,9 +737,23 @@ export abstract class FluidDataStoreContext
 	 * @param fromPath - The absolute path of the node that added the reference.
 	 * @param toPath - The absolute path of the outbound node that is referenced.
 	 * @param messageTimestampMs - The timestamp of the message that added the reference.
+	 * @param fromPkg - The package path of the node that adds the reference.
+	 * @param toPkg - The package path of the node to which reference is added.
 	 */
-	public addedGCOutboundRoute(fromPath: string, toPath: string, messageTimestampMs?: number) {
-		this.parentContext.addedGCOutboundRoute(fromPath, toPath, messageTimestampMs);
+	public addedGCOutboundRoute(
+		fromPath: string,
+		toPath: string,
+		messageTimestampMs?: number,
+		fromPkg?: readonly string[],
+		toPkg?: readonly string[],
+	) {
+		this.parentContext.addedGCOutboundRoute(
+			fromPath,
+			toPath,
+			messageTimestampMs,
+			fromPkg ?? this.pkg,
```

Comment:
Should we just always use `this.pkg`? When would the provided value be different?

Reply:
Actually now I'm wondering the flip side - why is this a legitimate default?  Brain is slowing down so maybe it'll be obvious next time I look at this...

Reply:
The only usage I can see for DataStoreContext is when processing a DDS attach message, and neither pkg is passed.  This is fine, ContainerRuntime will fill them in.  I would just not even include them in the signature of this implementation - it will still satisfy the interface... although it means that if someone _does_ pass them they'll be ignored.  Hmm I guess it's ok to pass them through, but it feels cleaner somehow to not and just say ContainerRuntime will compute them.

Adding this fallback here feels more confusing - what scenario is it actually helpful?  If someone passes it, use it. If not, fall back to the common computation in ContainerRuntime.  Less code/cases is better.

Thoughts?

Reply:
Yeah I have that doubt too. In `addedGCOutboundRoute` to begin with. We accept both fromPath and fromPkg, and eventually do this
> const packagePath = toPkg ?? this.getGCNodePackagePath(toPath);

Curious what's the advantage of also allowing caller to send the frompkg and topkg, when we already have the frompath and to path!

Reply:
I guess to keep things simple, we could leave `addedGCOutboundRoute` unchanged and just have container runtime get the package path when calling GarbageCollector. The goal was to not have the GarbageCollector try to get it.

The reason I added package paths everywhere is for optimization, i.e., if the call is coming directly from a data store or when processing data store attach message, the package path is readily available so there is no need to get it again. But getting the package is cheap enough so we should not add this complication.